bundles/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/handler/IPBridgeThingHandler.java

```diff
@@ -70,20 +110,64 @@ public void initialize() {
         String ip = config.getIpAddress();
         InetSocketAddress localEndPoint = null;
         boolean useNAT = false;
-        int ipConnectionType;
+
+        IPClient.IpConnectionType ipConnectionType;
         if (MODE_TUNNEL.equalsIgnoreCase(connectionTypeString)) {
             useNAT = config.getUseNAT();
-            ipConnectionType = CustomKNXNetworkLinkIP.TUNNELING;
+            ipConnectionType = IPClient.IpConnectionType.TUNNEL;
+        } else if (MODE_SECURE_TUNNEL.equalsIgnoreCase(connectionTypeString)) {
+            useNAT = config.getUseNAT();
+            ipConnectionType = IPClient.IpConnectionType.SECURE_TUNNEL;
+
+            if (!securityAvailable) {
+                logger.warn("Bridge {} missing security configuration for secure tunnel", thing.getUID());
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "Security configuration missing for secure tunnel");
+                return;
+            }
+            boolean tunnelOk = ((secureTunnel.user > 0) && (secureTunnel.devKey.length == 16)
+                    && (secureTunnel.userKey.length == 16));
+            if (!tunnelOk) {
+                logger.warn("Bridge {} incomplete security configuration for secure tunnel", thing.getUID());
```

Comment:
Could you add specific checks for the parameters, so that you could directly hint to the user, which parameter he forgot to provide?

Reply:
Not sure if I should add individual checks.... This is anyway an advanced setup and requires reading the manual.
The next PR for data sec will introduce a second way to configure the items;: the keyring.
If you are fine with it, I would leave it as it is for now.

Reply:
Ok for me.